File: superReducedString_hacker_rank.rb
def superReducedString(s)
    removed = false

    while !removed
    removed = true
        i = 0
        while i < s.length-1
            if s[i] == s[i+1]
                removed = false
                s = s[0...i] + s[i+2..-1]
            end
            i += 1
        end
    end
    if s.length > 0
        return s
    else
        return 'Empty String'
    end
    
end

File: superReducedString_hacker_rank.js
function superReducedString(s) {
    let sorted = false;
    while (!sorted) {
        sorted = true;
        let i = 0;
        while (i < s.length - 1) {
            if (s[i] == s[i + 1]) {
                sorted = false;
                s = s.slice(0, i) + s.slice(i + 2);
            }

            i++
        }
    }
    if (s.length) {
        return s;
    } else {
        return "Empty String";
    }
}